src/org-units-selector/OrgUnitsSelector.jsx

```diff
@@ -52,45 +46,48 @@ export default class OrgUnitsSelector extends React.Component {
     };
 
     static childContextTypes = {
-        d2: PropTypes.object.isRequired,
+        api: PropTypes.object.isRequired,
     };
 
     constructor(props) {
         super(props);
 
         this.state = {
-            d2: props.d2,
+            cancel: null,
             levels: null,
             roots: null,
             groups: null,
             currentRoot: null,
         };
         this.contentsStyle = { ...styles.contents, height: props.height };
+    }
 
+    componentDidMount() {
+        const { props } = this;
```

Comment:
I moved this code from constructor, I was getting a warning on the console about calls to setState on non-mounted components. The original dhis2 code was wrong, no request must be made in a constructor.